Extract location data from Json, and display them on map

ElementMo opened the pull request.

ElementMo committed:
Extract location data from Json, and display them on map

EgMan commented on load-json.js:
```diff
@@ -2,26 +2,45 @@ var loadedJSON = null;
+
+            // create loc data array
+            var locationData = [];
+            for(var i=0; i<loadedJSON.locations.length; i++){
```
This should fail with an alert message if there is no "locations" array or if the "locations" array is not of the correct format.

ElementMo replied:
Fixed.

EgMan commented on load-json.js:
```diff
@@ -2,26 +2,45 @@ var loadedJSON = null;
+            newOption = {
+                series: [
+                    {
+                        // data: [
```
remove unneeded code

ElementMo replied:
Deleted.

EgMan commented on load-json.js:
```diff
@@ -2,26 +2,45 @@ var loadedJSON = null;
+                ]
+            }
+            // update option
+            echartslayer.chart.setOption(newOption);
```
Is it possible to dynamically/periodically load these points to the map as we parse through the json object? This way, we don't have to see just a loading screen, we could have the points populate as we process them.

EgMan replied:
future enhancement

EgMan commented on load-json.js:
```diff
@@ -2,26 +2,45 @@ var loadedJSON = null;
+                ]
+            }
+            // update option
+            echartslayer.chart.setOption(newOption);
```
Nice, this works, I tried it.  That being said, the points disappear and re-render every time you move the map and it's quite slow.  Is there any way around this?

ElementMo replied:
There is a way to do this. It requires echarts-gl extention, I'll update this later.

EgMan replied:
+1

EgMan commented on load-map.js:
```diff
@@ -1,3 +1,5 @@
+var echartslayer;
+
 var map = new mapboxgl.Map(
```
After playing around with my location map for a bit, I realized that we should maybe display city names or something so it's easy to have bearings on where we are.  If this is possible, we can make this a future enhancement.

No resolution needed.

ElementMo replied:
Okay, I'll dig into that.

EgMan replied:
+1

ElementMo committed:
Fix problems

1. Add json data content error when there is no required json keywords;
2. Deleted useless code;

ElementMo committed:
uploatlocfile
